src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -0,0 +1,145 @@
+import { StatisticsService } from 'app/shared/statistics-graph/statistics.service';
+import { ChartDataSets, ChartOptions, ChartType } from 'chart.js';
+import { BaseChartDirective, Label } from 'ng2-charts';
+import { DataSet } from 'app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component';
+import { TranslateService } from '@ngx-translate/core';
+import { Graphs, SpanType } from 'app/entities/statistics.model';
+import { CourseManagementStatisticsModel } from 'app/entities/quiz/course-management-statistics-model';
+
+@Component({
+    selector: 'jhi-statistics-average-score-graph',
+    templateUrl: './statistics-average-score-graph.component.html',
+})
+export class StatisticsAverageScoreGraphComponent implements OnInit {
+    @Input()
+    exerciseAverageScores: CourseManagementStatisticsModel[];
+    @Input()
+    courseAverage: number;
+
+    // Html properties
+    LEFT = false;
+    RIGHT = true;
+    SpanType = SpanType;
+    Graphs = Graphs;
+
+    // Histogram related properties
+    barChartOptions: ChartOptions = {};
+    barChartType: ChartType = 'bar';
+    lineChartType: ChartType = 'line';
+    exerciseAverageLegend: string;
+    courseAverageLegend: string;
+    chartName: string;
+    barChartLegend = true;
+
+    // Data
+    barChartLabels: Label[] = [];
+    chartData: ChartDataSets[] = [];
+    dataForSpanType: number[];
+
+    // Left arrow -> decrease, right arrow -> increase
+    currentPeriod = 0;
+
+    @ViewChild(BaseChartDirective) chart: BaseChartDirective;
+
+    constructor(private service: StatisticsService, private translateService: TranslateService) {}
+
+    ngOnInit(): void {
+        this.chartName = this.translateService.instant(`artemisApp.course.averageScore`);
+        this.exerciseAverageLegend = this.translateService.instant('artemisApp.courseStatistics.exerciseAverage');
+        this.courseAverageLegend = this.translateService.instant('artemisApp.courseStatistics.courseAverage');
+        this.initializeChart();
+        this.createCharts();
+    }
+
+    private initializeChart(): void {
+        this.barChartLabels = this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod).map((exercise) => exercise.exerciseName);
+        this.chartData = [
+            {
+                // Average course score line
+                label: this.courseAverageLegend,
+                data: new Array(this.barChartLabels.length).fill(this.courseAverage),
+                backgroundColor: 'rgba(93,138,201,1)',
+                fill: false,
+                pointBackgroundColor: 'rgba(93,138,201,0)',
+                pointBorderColor: 'rgba(93,138,201,0)',
+                pointHoverBackgroundColor: 'rgba(93,138,201,1)',
+                pointHoverBorderColor: 'rgba(93,138,201,0)',
+                borderColor: 'rgba(93,138,201,1)',
+                hoverBackgroundColor: 'rgba(93,138,201,1)',
+                hoverBorderColor: 'rgba(93,138,201,1)',
```

Comment:
It might be better to put colors into variables so that it is easier to see which are the same color, also to change it.

Reply:
👍